specs/impact-graph.md

```diff
@@ -0,0 +1,223 @@
+
+### Example
+
+Concrete example of how we might model the carbon impact of a web server using an **Impact Graph**.
+
+```
+My Web Server Impact Graph
+```
+
+We first create an **Impact Graph** that describes the ontology we want to measure. This might be an application like this case which is a Web Server, but it also might be a campaign or even an individual user journey. It could even be used to measure the embodied carbon of physical devices. What we are proposing here isn't limited to just software.
+
+```
+My Web Server Impact Graph
+├── Backend Server
+├── Load Balencer
+└── Cache Servers
+```
+
+We second break out the ontology into the nodes which we want to measure. The important thing is that we are breaking things out not as concepts, but as seperate and distint measurable components of the whole. 
+
+An **Impact Graph Node (IGN)** is something we want to measure distinct from every other node in the Impact Graph. That's it, we can clump all the servers together into one IGN regardless of what's running on them IF we want to measure them all together. We can break it up into one IGN per-process IF we want to measure per-process. The IGN describes an atomic thing we want to measure, that's it.
+
+What this means is that although a component might be "Backend Servers" since we'd use different approaches to measure "Intel Backend Servers" and "AMD Backend Servers" we might actually break it out into two IGNs, "AMD Backend Servers" and "Intel Backend Servers". This sounds complex but since our goal is to measure, measuring dictates how we strucutre the graph.
+
+```
+My Web Server Impact Graph
+├── Backend Server <- class BoavistaVM extends IMI -> Boavista IMM
+├── Load Balencer <- class BoavistaVM extends IMI -> Boavista IMM
+└── Cache Servers <- class IntelVM extends IMI -> Intel IMM
+```
+
+The next step is figuring out which **Impact Measurement Models** we want to use to measure each **Impact Graph Node**. In the above example we want to use Boavista's **Impact Measurement Model** to measure the Backend Server and Load Balencer because those are runing on a mixture of different types of underlying chips. The cache servers might however only be running on Intel Chips and therefore we may want to use Intels IMM since it's more accurate for Intel products. To interact with each IMM we use an **Impact Measurement Plugin** which is just a class that extends the **Impact Measurement Interface** and implements the functions in the interface.
+
+If later on we wanted to use another IMM, perhaps the Cloud Carbon Footprint IMM, we would 
+
+1. install the CCF package
+2. import it
+3. use the CCF class instead of the Boavista class.
+
+```
+My Web Server Impact Graph
+├── Backend Server <- class CCFVM extends IMI -> CCF IMM
+├── Load Balencer <- class CCFVM extends IMI -> CCF IMM
+└── Cache Servers <- class IntelVM extends IMI -> Intel IMM.
+```
+
+
+All IMPs expose the same class Interface (IMI) so can be swapped in and out easily.

```

Comment:
All IMPs extend from the same base class Interface (IMI), so can be swapped in and out easily. New IMPs can be added by Organizations/Users by extending the base class Interface.

Reply:
```suggestion
All IMPs extend from the same base class Interface (IMI), so can be swapped in and out easily. New IMPs can be added by Organizations/Users by extending the base class Interface.
```